### PR TITLE
Feature/NIJO-1085 split MarkupPolicyManager service for agents and admins

### PR DESCRIPTION
**Pull request by DFSko**

### CODE REVIEW

File: Api/Services/Markups/AdminMarkupPolicyManager.cs [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+
+            async Task<Result<MarkupPolicy>> GetPolicy()
+            {
+                var policy = await _context.MarkupPolicies.SingleOrDefaultAsync(p => p.Id == policyId);
+                return policy == null ? Result.Failure<MarkupPolicy>("Could not find policy") : Result.Success(policy);
+            }
+
+
+            async Task<Result> DeletePolicy(MarkupPolicy policy)
```
Comment (mingazhev) [inline Api/Services/Markups/AdminMarkupPolicyManager.cs:74]: This method never returns failed result

File: Api/Services/Markups/AdminMarkupPolicyManager.cs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+                _context.Update(policy);
+                await _context.SaveChangesAsync();
+                return Result.Success();
+            }
+        }
+
+
+        public async Task<Result<List<MarkupPolicyData>>> Get(MarkupPolicyScope scope)
+        {
```
Comment (mingazhev) [inline Api/Services/Markups/AdminMarkupPolicyManager.cs:114]: This method never returns failure result

File: Api/Services/Markups/AgentMarkupPolicyManager.cs [hunk 3]
```diff
@@ -78,10 +78,7 @@ public Task<Result> Remove(int policyId)
             async Task<Result<MarkupPolicy>> GetPolicy()
             {
                 var policy = await _context.MarkupPolicies.SingleOrDefaultAsync(p => p.Id == policyId);
-                if (policy == null)
-                    return Result.Failure<MarkupPolicy>("Could not find policy");
-
-                return Result.Success(policy);
+                return policy == null ? Result.Failure<MarkupPolicy>("Could not find policy") : Result.Success(policy);
```
Comment (mingazhev) [inline Api/Services/Markups/AgentMarkupPolicyManager.cs:81]: :Please, place each expression on the separate line, it'll be much easier to read:
```
return policy == null 
? Result.Failure<MarkupPolicy>("Could not find policy") 
: Result.Success(policy);
```

Comment (mingazhev) [reply Api/Services/Markups/AgentMarkupPolicyManager.cs:81]: Here and in other places

File: Api/Services/Markups/MarkupPolicyManagerFactory.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public class MarkupPolicyManagerFactory : IMarkupPolicyManagerFactory
+    {
+        public MarkupPolicyManagerFactory(IServiceProvider serviceProvider)
+        {
+            _serviceProvider = serviceProvider;
+        }
+
+
+        public IMarkupPolicyManager Get(MarkupPolicyManagerTypes type)
```
Comment (mingazhev) [inline Api/Services/Markups/MarkupPolicyManagerFactory.cs:14]: Factory looks overhead for that.
We do not hide anything from the client of interface there, it is used in different separate parts of the application.
Injecting separate interfaces will be better

File: Api/Services/Markups/AgentMarkupPolicyManager.cs [hunk 5]
```diff
@@ -15,9 +15,9 @@
 
 namespace HappyTravel.Edo.Api.Services.Markups
 {
-    public class MarkupPolicyManager : IMarkupPolicyManager
+    public class AgentMarkupPolicyManager : IMarkupPolicyManager
     {
-        public MarkupPolicyManager(EdoContext context,
+        public AgentMarkupPolicyManager(EdoContext context,
             IAgentContextService agentContextService,
```
Comment (mingazhev) [inline Api/Services/Markups/AgentMarkupPolicyManager.cs:21]: Please, inject agentContextService to controllers only

**Review by mingazhev — CHANGES_REQUESTED** [review]

File: Api/Services/Markups/AdminMarkupPolicyManager.cs (hunk 1 above)
Comment (DFSko) [reply Api/Services/Markups/AdminMarkupPolicyManager.cs]: Fixed

File: Api/Services/Markups/AdminMarkupPolicyManager.cs (hunk 2 above)
Comment (DFSko) [reply Api/Services/Markups/AdminMarkupPolicyManager.cs:115]: Fixed

File: Api/Services/Markups/AgentMarkupPolicyManager.cs (hunk 3 above)
Comment (DFSko) [reply Api/Services/Markups/AgentMarkupPolicyManager.cs]: Fixed

File: Api/Services/Markups/MarkupPolicyManagerFactory.cs (hunk 4 above)
Comment (DFSko) [reply Api/Services/Markups/MarkupPolicyManagerFactory.cs]: Fixed

File: Api/Services/Markups/AgentMarkupPolicyManager.cs (hunk 5 above)
Comment (DFSko) [reply Api/Services/Markups/AgentMarkupPolicyManager.cs]: Fixed

**Review by mingazhev — APPROVED** [review]

**Review by kirillta — APPROVED** [review]

**Pull request merged**